src/lib/block_if.c

```diff
@@ -712,9 +712,11 @@ blockif_open(const char *optstr, const char *ident)
 		}
 		/* Lock the file to prevent concurrent write+write or read+write as this
 		   would usually lead to corruption */
-		if (flock(fd, LOCK_NB | (ro ? LOCK_SH : LOCK_EX)) < 0) {
-			perror("Could not lock backing file");
-			goto err;
+		if (!ro) {
+			if (flock(fd, LOCK_NB | (ro ? LOCK_SH : LOCK_EX)) < 0) {
```

Comment:
There is a now redundant `ro ?` within this check. But I'm not sure about this change. It seems to me that a shared lock is the correct thing for a ro file since while multiple readers are ok you want to exclude any writers from coexisting with those readers. Having `ro` use `LOCK_SH` and writers use `LOCK_EX` (i.e. the status quo) ought to achieve this I think

It seems to me that the existing code was trying to do what we want &mdash; if it is broken it'd be nice to know why, in any case I don't think removing all locking from the `ro` case is likely to be what we really want.

Reply:
I agree, the previous code does work.  All it needs is `ro` to be set.